util/hook/hook.go

```diff
@@ -6,50 +6,50 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
 	"github.com/argoproj/argo-cd/common"
-	argoappv1 "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	. "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
 )
 
-// IsHook indicates if the object is either a Argo CD or Helm hook
 func IsHook(obj *unstructured.Unstructured) bool {
```

Comment:
We should not remove the helm hook detection. All helm hooks need to be completely ignored, for the time being, with the exception of the crd-hook.

Reply:
I've moved the ignore code into the sync itself. Otherwise, Helm hooks appear as normal resources.

Reply:
OK. Totally ignore Helm hooks.